src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -471,7 +471,7 @@ impl crate::CommandRunner for CommandRunner {
     {
       let command_runner = self.clone();
       let result = result.clone();
-      let _write_join = self.executor.spawn(in_workunit!(
+      let write_fut = self.executor.spawn(in_workunit!(
```

Comment:
I think that a tail task should likely be added by an explicit method of `Executor`, for two reasons:
1. If the task is not `spawn`d, it would likely be an error, because the work would not even start until shutdown. That's would potentially be neat to hack, because you could queue up lazy work until shutdown, but IMO you would want that to be explicit.
2. Being able to poll `is_finished` on a `JoinHandle` returned directly by the `Executor`, or a wrapper type if need be would be very helpful in terms of the debug output at the end of the run (see other comment).

Reply:
Is the executor associated 1:1 with a session?

Reply:
No, it isn't: an executor may be used across multiple sessions. Hence passing the `tail_tasks` as an argument in the snippet: `executor.spawn_tail(&mut tail_tasks, fut)`.